Playwright Visual Snapshot Comparision Test

suresh-gangumalla opened the pull request:
Playwright-based visual regression tests have been added, including a sample example application and a visual-tests project that supports two execution modes: Local Mode and CI (Docker) Mode. To get started, first install the Blits dependencies

```
npm install
cd examples && npm install
cd visual-tests && npm install
```

To run tests in Local Mode, begin by capturing baseline snapshots
```npm run test:visual -- -- --capture```

Then execute the visual snapshot comparison
```npm run test:visual```

For CI (Docker) Mode, build the Docker image
```npm run build:docker```

And run the visual regression tests in CI mode
```npm run test:visual -- -- --ci```

suresh-gangumalla committed:
Playwright snap capture & compare test

Signed-off-by: suresh-gangumalla <sureshkumar.gangumalla@invecas.com>

suresh-gangumalla committed:
Merge remote-tracking branch 'upstream/dev' into feat/snap-comp-playwright

Signed-off-by: Suresh Kumar Gangumalla <sureshkumar.gangumalla@invecas.com>

gsk-work committed:
Resolved issues in docker container visual tests

Signed-off-by: Suresh Kumar Gangumalla <sganguma@cadence.com>

michielvandergeest committed:
Added missing playwright dev dependency.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lightning-js/blits?pullRequest=537) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/lightning-js/blits?pullRequest=537) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: michielvandergeest<br/>:white_check_mark: suresh-gangumalla<br/>:x: gsk-work<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lightning-js/blits?pullRequest=537) it.</sub>

suresh-gangumalla committed:
Updated playwright image to nobel version

Earlier playwright version was build on top of linux 22.x
and latest changes to msdf-font-generator linux binaries requires
GLIBC > 2.38. Ubuntu 24.X comes with GLIBC version which satisifies
msdf-font-generator binaries in linux OS so updated playwright to nobel
version which build on Ubutntu 24.X.

Masking node_modules of examples and visual-tests projects inside
docker from blits directory complete mount so that examples and
visual-tests projects get node_modules of playwright image (linux)
node-modules on container run instead of depending on host machine
installed node-modules

Signed-off-by: suresh-gangumalla <sureshkumar.gangumalla@invecas.com>

suresh-gangumalla committed:
Added shaders example for visual test

Signed-off-by: suresh-gangumalla <sureshkumar.gangumalla@invecas.com>
